pkg/apis/v1beta1/nodeclass_validation.go

```diff
@@ -97,10 +99,15 @@ func (in *NodeClassSpec) validateSecurityGroupSelectorTerms() (errs *apis.FieldE
 	return errs
 }
 
+//nolint:gocyclo
 func (in *SecurityGroupSelectorTerm) validate() (errs *apis.FieldError) {
 	errs = errs.Also(validateTags(in.Tags).ViaField("tags"))
 	if len(in.Tags) == 0 && in.ID == "" && in.Name == "" {
 		errs = errs.Also(apis.ErrGeneric("expect at least one, got none", "tags", "id", "name"))
+	} else if len(in.Tags) > 0 && in.ID != "" || in.Name != "" && in.ID != "" {
+		errs = errs.Also(apis.ErrGeneric(`"id" is mutually exclusive, cannot be set with a combination of other fields in`))
+	} else if len(in.Tags) > 0 && in.Name != "" || in.ID != "" && in.Name != "" {
+		errs = errs.Also(apis.ErrGeneric(`"name" is mutually exclusive, cannot be set with a combination of other fields in`))
```

Comment:
I think it's cleaner to start with the common condition and to || the differences. If not, I think it's worth making a new block:
```suggestion
	} else if in.ID != "" && (len(in.Tags) > 0 || in.Name != "") {
		errs = errs.Also(apis.ErrGeneric(`"id" is mutually exclusive, cannot be set with a combination of other fields in`))
	} else if in.Name != "" && (len(in.Tags) > 0 || in.ID != "") {
		errs = errs.Also(apis.ErrGeneric(`"name" is mutually exclusive, cannot be set with a combination of other fields in`))
```

or 

```suggestion
	} else if in.ID != "" {
	     if len(in.Tags) > 0 || in.Name != "" {
		    errs = errs.Also(apis.ErrGeneric(`"id" is mutually exclusive, cannot be set with a combination of other fields in`))
	     }	
	} else if in.Name != "" {
	    if len(in.Tags) > 0 || in.ID != "" {
		errs = errs.Also(apis.ErrGeneric(`"name" is mutually exclusive, cannot be set with a combination of other fields in`))
		}
```